pkg/modules/metricsmodule/metricsmodule.go

```diff
@@ -0,0 +1,22 @@
+package metricsmodule
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/SigNoz/signoz/pkg/types/metricsmoduletypes"
+	"github.com/SigNoz/signoz/pkg/valuer"
+)
+
+// Handler exposes HTTP handlers for the metrics module.
+type Handler interface {
+	GetStats(http.ResponseWriter, *http.Request)
+	GetTreemap(http.ResponseWriter, *http.Request)
+}
+
+// Module represents the metrics module interface.
+type Module interface {
+	GetStats(ctx context.Context, orgID valuer.UUID, req *metricsmoduletypes.StatsRequest) (*metricsmoduletypes.StatsResponse, error)
+	GetTreemap(ctx context.Context, orgID valuer.UUID, req *metricsmoduletypes.TreemapRequest) (*metricsmoduletypes.TreemapResponse, error)
+	GetUpdatedMetricsMetadata(ctx context.Context, orgID valuer.UUID, metricNames []string) (map[string]*metricsmoduletypes.MetricMetadata, error)
```

Comment:
Module shouldn't have `GetUpdatedMetricsMetadata`. Look at it as this, the module will be consumed by anyone, so when they read it shouldn't require them making more questions. Why is it a `GetUpdatedMetricsMetadata` instead of `GetMetricsMetadataMulti` where it doesn't really matter to the caller the updated context of the metric and `-Multi` to indicate this takes multiple metric names as input.